### ISSUE TITLE
[BUG/Help] <title>ptuning过程中，训练数据集格式问题

### ISSUE THREAD

**Issue by znzjugod** [THUDM/ChatGLM2-6B#349]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

我的train.json文本是这样的：
{"labels": "请按照规定格式介绍苏州。", "predict": "{ "country": "中国","city": 苏州}"}
使用这样的训练集进行训练，则会报错，报错内容如下:
JSON parse error: Column(/predict) changed from string to object

### Expected Behavior

能够正常解析json文本

### Steps To Reproduce

导入数据集后，执行bash train.sh

### Environment

```markdown
- OS:5.10.0-136.12.0.86.oe2203sp1.x86_64
- Python:3.9
- Transformers:4.29.1
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by zobnechopin** [comment]
> 

看起来像是 predict 内容格式问题？

另外很多行数据的话应该长这个样子：

{"title": "summarised text","body": "text to be summarized"}
{"title": "summarised text","body": "text to be summarized"}

**Comment by znzjugod** [comment]
> 

是的，按你这种写法是没问题的。
但是我想达到的效果是，输入prompt后，可以返回一个json文本。
所以报错也是predict的类型从str变为object。
但是代码中我没看到哪里限制了只能读str。
predict的内容如果是：{"labels": "请按照规定格式介绍苏州。", "predict": "["country": "中国","city": 苏州]"}
也会报错。

所以是只能接str类型的内容吗？

**Comment by znzjugod** [comment]
> > 
> 
> 看起来像是 predict 内容格式问题？
> 
> 另外很多行数据的话应该长这个样子：
> 
> {"title": "summarised text","body": "text to be summarized"} {"title": "summarised text","body": "text to be summarized"}

是的，按你这种写法是没问题的。
但是我想达到的效果是，输入prompt后，可以返回一个json文本。
所以报错也是predict的类型从str变为object。
但是代码中我没看到哪里限制了只能读str。
predict的内容如果是：{"labels": "请按照规定格式介绍苏州。", "predict": "["country": "中国","city": 苏州]"}
也会报错。

所以是只能接str类型的内容吗？

**Comment by kuang1996** [comment]
应该是需要转义，也就是这样的格式：{"labels": "请按照规定格式介绍苏州。", "predict": "[\"country\":\ "中国\",\"city\": \"苏州\"]"}
这样引号才能匹配。
我数据集是这样的格式，训练出来结果就是一个json。没有问题。

**Comment by zobnechopin** [comment]
> > 看起来像是 predict 内容格式问题？
> > 另外很多行数据的话应该长这个样子：
> > {"title": "summarised text","body": "text to be summarized"} {"title": "summarised text","body": "text to be summarized"}
> 
> 是的，按你这种写法是没问题的。 但是我想达到的效果是，输入prompt后，可以返回一个json文本。 所以报错也是predict的类型从str变为object。 但是代码中我没看到哪里限制了只能读str。 predict的内容如果是：{"labels": "请按照规定格式介绍苏州。", "predict": "["country": "中国","city": 苏州]"} 也会报错。
> 
> 所以是只能接str类型的内容吗？

转义可以使用，或者 JSON String 来回处理也问题不大

**Comment by znzjugod** [comment]
转义后可以正常训练了。多谢大家的答复

**Comment by znzjugod** [comment]
转义后可以正常训练了。多谢大家的答复
